### PR TITLE
Tweak: Added editor_cta=top_bar parameter when connecting through the top bar [ED-4459]

### PR DESCRIPTION
**Pull request by nisan250**

### CODE REVIEW

File: core/common/modules/connect/apps/base-app.php [hunk 1]
```diff
@@ -533,9 +533,19 @@ protected function get_remote_authorize_url() {
 			'redirect_uri' => rawurlencode( $redirect_uri ),
 			'may_share_data' => current_user_can( 'manage_options' ) && ! Tracker::is_allow_track(),
 			'reconnect_nonce' => wp_create_nonce( $this->get_slug() . 'reconnect' ),
-		], $this->get_remote_site_url() );
+		];
 
-		return $url;
+		$propagate_query_args_whitelist = [
```
Comment (Nevoss) [inline core/common/modules/connect/apps/base-app.php:538]: @matipojo we decide to do this as an array to support UTM query params.

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 2]
```diff
@@ -265,6 +265,26 @@ public function test_http_request__with_connect() {
...
+		$query_args = [];
+
+		$parsed_url = wp_parse_url( $url );
+		parse_str( $parsed_url['query'], $query_args );
+
+		// Assert
+		$this->assertFalse( isset( $query_args['black_list'] ) );
+		$this->assertTrue( isset( $query_args['editor_cta'] ) );
+		$this->assertEquals( 'some_action', $query_args['editor_cta'] );
+
```
Comment (Nevoss) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:285]: Remove empty line

**Review by Nevoss — CHANGES_REQUESTED** [review]

**Review by Nevoss — APPROVED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 3]
```diff
@@ -265,6 +265,25 @@ public function test_http_request__with_connect() {
 		$this->assertEquals( (object) [ 'status' => 'success' ], $result );
 	}
 
+	public function test_get_remote_authorize_url__propagate_query_args() {
+		// Arrange
+		$_REQUEST['black_list'] = 'test';
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:270]: ?

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 4]
```diff
@@ -265,6 +265,25 @@ public function test_http_request__with_connect() {
...
+		// Act
+		$url = $this->app_stub->proxy_get_remote_authorize_url();
+
+		$query_args = [];
+
+		$parsed_url = wp_parse_url( $url );
+		parse_str( $parsed_url['query'], $query_args );
+
+		// Assert
+		$this->assertFalse( isset( $query_args['black_list'] ) );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:282]: ?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (nisan250) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:270]: When we pass a value that does not exist in $propagate_query_args_whitelist, the test will return false.

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 4 above)
Comment (nisan250) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:282]: same as above

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (matipojo) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:270]: Maybe change to `unknown_arg`

**Review by matipojo — CHANGES_REQUESTED** [review]
PHP Lint

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (nisan250) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:270]: done

**Review by matipojo — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/modules/connect/apps/base-app.php [hunk 5]
```diff
@@ -533,9 +533,19 @@ protected function get_remote_authorize_url() {
...
+			'editor_cta',
+		];
+
+		foreach ( $propagate_query_args_whitelist as $key ) {
+			if ( isset( $_REQUEST[ $key ] ) ) {
+				$query_args[ $key ] = $_REQUEST[ $key ];
```
[the remark this answers is not in the document]
Comment (nisan250) [reply core/common/modules/connect/apps/base-app.php]: Done

**Comment by Nevoss** [comment]
@matipojo Please close this is PR.
ref: https://github.com/elementor/elementor/pull/15998

**Pull request closed without merging**